lib/liquid/context.rb

```diff
@@ -62,10 +63,10 @@ def pop_interrupt
       @interrupts.pop
     end
 
-    def handle_error(e, token = nil)
+    def handle_error(e, line_number = nil)
       if e.is_a?(Liquid::Error)
-        e.template_name = template_name
-        e.set_line_number_from_token(token)
+        e.template_name ||= template_name
```

Comment:
because that is what set_line_number_from_token was doing so that the line number isn't replaced when an exception is re-raised multiple times as was happening in BlockBody